### DISCUSSION TITLE
Avoid motorways with motorroad=yes

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#666] [Q&A] [answered]
There is the tag `motorroad=yes` in OpenStreetMap database.

More details: https://wiki.openstreetmap.org/wiki/Key:motorroad

The [driving profiles](https://github.com/poutnikl/Brouter-profiles/tree/master/CarProfiles) by @poutnikl use this tag in the **avoid motorways** option.

Should we also use it in Cruiser driving profile in the **avoid motorways** option?

**Comment by bios4** [comment]
I would vote yes for that proposal.
At least in Austria this "motorroad" classification is fully applicable to "avoid motorways"...

**Comment by DolomitesTom** [comment]
Yes with the 5 gradations.

**Comment by devemux86** [comment]
> Should we also use it in Cruiser driving profile in the **avoid motorways** option?

Available in **Cruiser 5.0.2-b5**.

**Comment by Leeb64** [comment]
Would it be possible to distinguish between toll roads (Switzerland (vignette)) and normal motorway tolls (France)?

**Reply by devemux86** [reply]
Routing works with the OpenStreetMap tags. What tags do you mean?

Also routing is global for all countries.

**Reply by Leeb64** [reply]
If you want to drive on Swiss or Austrian motorways you need a vignette.
You buy this thing and can use all highways (for a year/month).
In France I basically pay for every kilometer I drive.
I don't know whether the OSM maps make this distinction.

**Reply by MotoUKRider** [reply]
@Leeb64 Waking this thread up because I was curious (EU tour planning etc), too.

Both French and Swiss motorways use `toll=yes` in their properties. It's a bit "one size fits all" so there's no real distinction between a road with a payment booth or having to display a sticker... unless you spot a barrier/booth/machine upon entry (like on the French network).

You can examine any bit of OpenStreetMap in your browser, just zoom into the area, right-click and select "Query features", moving your mouse over the list of objects in the left-hand column will highlight them as you move, you can then click one to see the tags.

It is also possible to [enhance the toll features](https://wiki.openstreetmap.org/wiki/Key:toll?uselang=en), so where *most* vehicles have to pay, but motorcycles are free, it should have `toll=yes` and `toll:motorcycle=no` although this may not be supported by all routing engines (BRouter currently struggles with this).

**Comment by Ciutnik** [comment]
If I understand correctly, the exclusion of motorways will affect other roads tagged `motorroad=yes`? 

If so, it is probably not a good solution. In my area the vast majority of `trunk` is tagged as `motorroad`. So they are excluded on a par with `motorway`. It can be confusing!

But even worse, I also found `primary` and even `secondary` tagged `motorroad=yes`(!). I suggest you to check your neighbourhoods by simple queries on overpass-turbo.eu: e.g. 
`highway!=trunk AND highway!=motorway AND highway!=trunk_link AND highway!=motorway_link AND motorroad=yes IN [your area]` 
This will showcase non-motorways and non-trunks (something doesn't work out for me using regular expressions for "_link" 🙂) tagged as `motorroad`.

**Reply by devemux86** [reply]
We should all study the OpenStreetMap detailed description and test what happens.
Then we can decide if we really want it or not.

Remember that @poutnikl profiles (used by many) have it in avoid motorways.

**Reply by devemux86** [reply]
You can also test it on BRouter website.

**Reply by Ciutnik** [reply]
I tested on Cruiser b2 vs b6. With the **motorways** avoidance setting set to 5/5, the rest of the roads set to OFF. 

It is obvious that a **primary** category road (red 20) has been excluded in b6. Even if this is a rare case, it shouldn't be like this.

5.0.2-b2 vs -b6

<img src="https://github.com/user-attachments/assets/92f07140-459b-410b-9b46-567588937ac8" width="49%"/> <img src="https://github.com/user-attachments/assets/dde4deb2-2ee7-4233-90f4-9f20fc45d1e7" width="49%"/>

**Comment by devemux86** [comment, marked as answer]
As it can be confusing to mix different types of roads, let's not use this tag.
